worker/metrics/spool/metrics.go

```diff
@@ -55,6 +56,10 @@ func (f *metricFile) Close() error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	// If the file contents are garbage, don't try and use it.
+	if f.encodeErr != nil {
+		return nil
```

Comment:
shouldn't this be returning an error rather than nil?

Reply:
No because this is the Close(). The close itself worked and is done as cleanup as part of a defer.
The actual error is reported outside this. It's only here as a flag.